### PR TITLE
Implement support for sending custom headers for map resources

### PR DESCRIPTION
**Pull request by yavosh**
Add support for sending custom http headers from mapbox sdk 
Android only in this iteration

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// Top-level build file where you can add configuration options common to all sub-projects/modules.
```
Comment (mfazekas) [inline android/build.gradle]: I don't think this is related to custom headers

**Comment by mfazekas** [comment]
@yavosh  thanks for the PR, looks good to me. Can you also add usecase to the PR-s description?
Are you working on an ios version too?!

File: android/build.gradle (hunk 1 above)
Comment (kristfal) [reply android/build.gradle]: No, but it seems required for Android Studio to pick up the project properly

**Comment by kristfal** [comment]
@mfazekas Yavor is a colleague of mine. Yes, iOS will arrive in some form quite soon. The use case is to set custom headers, for instance if you are connecting to a custom tile server. 

I think this is good to merge. Docs will be added along with iOS.

Ex: MapboxGL. addCustomHeader('Authorization', ' Bearer <token>');

**Comment by henninghall** [comment]
@kristfal This is great! Any progress with the ios version? 😀

**Comment by Nopik** [comment]
When this will be implemented?

**Comment by yavosh** [comment]
Merged since `7.0.2`

**Pull request merged**